src/Couchbase.Lite.Tests.Shared/URLEndpointListenerTest.cs

```diff
@@ -862,20 +862,152 @@ public void TestStopListener()
 
         #endregion
 
-        #region Private Methods
+        #region Replicator Config Network Interface
 
-        private int GetEADDRINUSECode()
+        enum TestReplicatorNIType
+        {
+            ValidAddress_SERVER_REACHABLE,
+            ValidNI,
+            ValidNI_SERVER_UNREACHABLE,
+            InValidNI,
+            InValidAddress
+        }
+
+        #if !__ANDROID__ && !__IOS__ //Cannot run this test in emulators
```

Comment:
What about devices?

Reply:
Need to use the actual device. Emulators don't have Network Interface...

Reply:
Created an iOS ticket [CBL-3620](https://issues.couchbase.com/browse/CBL-3620)

Reply:
Test update to get NI tests work with Android device: https://github.com/couchbase/couchbase-lite-net/pull/1379/commits/92f15e260178e84b5bb188c977c6a000726824a1

Reply:
There should be nothing stopping testing on actual devices, but what is the issue with emulators?